power-management/rapl/powercap/runtest.sh

```diff
@@ -0,0 +1,230 @@
+    done
+}
+
+# XXX: This function is from power-management/rapl/powercap/test_capping.sh
+function test_capping
+{
+    # measurement time should be bigger than
+    # cat /sys/devices/virtual/powercap/intel-rapl/intel-rapl\:0/constraint_0_time_window_us
+    # which is 1 sec by default (at least on systems I have seen so far).
+    typeset measurement_time=30
+    typeset load_time=$(( measurement_time + 20 ))
+
+    # 1. no load, no cap
+    typeset e_no_no=$(get_energy $measurement_time)
+    rlLog "Power consumtion on uncapped system with no load: $e_no_no microwatts"
+
+    # 2. load, no cap
+    typeset cpus=$(lscpu | grep '^CPU(s):' | \
+                   sed 's/^CPU(s):[^0-9]*\([0-9]*\)/\1/')
+    loadm $load_time $cpus
+    typeset e_full_no=$(get_energy $measurement_time)
+    rlLog "Power consumtion on uncapped system with full load: $e_full_no microwatts"
+
+    # 3. cap to average between full load and no load
+    typeset e_mid=$(( (e_no_no + e_full_no) / 2 ))
+    rlLog "Going to cap the system to: $e_mid microwatts"
+
+    typeset old_cap=$(get_cap)
+    set_cap $e_mid
+    loadm $load_time $cpus
+    typeset e_full_cap=$(get_energy $measurement_time)
+    rlLog "Power consumtion on capped system with full load: $e_full_cap microwatts"
+
+    #
+    # measured_max is maximal measured value which will not cause fail
+    # measured_min is minimal measured value which will not cause fail
+    # *11/10 - add 10% in bash which works in integers only - we're
+    # using microwatts, so numbers are milions to hundreds of milions,
+    # so tehre is no problem with rounding error
+    # *9/10 - substract 10%
+    #
+    typeset measured_max=$(( e_mid * 11 / 10 ))
+    typeset measured_min=$(( e_mid *  9 / 10 ))
+
+    rlLog "capping back to $old_cap"
+    set_cap $old_cap
+
+    rlLog "e_no_no      = $e_no_no"
+    rlLog "e_full_no    = $e_full_no"
+    rlLog "e_mid        = $e_mid"
+    rlLog "e_full_cap   = $e_full_cap"
+    rlLog "measured_max = $measured_max"
+    rlLog "measured_min = $measured_min"
+    if (( e_full_cap > measured_max || e_full_cap < measured_min )); then
+        rlLog "FAIL: measured energy consumption doesn't match capped value"
+        return 1
+    fi
+
+    return 0
+}
+
+#
+# Check CPU vendor is Intel or not
+#
+function is_intel
+{
+    typeset vendor=$(grep vendor /proc/cpuinfo | sort -u | awk '{print $3}')
+    [[ $vendor == "GenuineIntel" ]] && return 0 || return 1
+}
+
+#
+# Check the system is kvm or not
+#
+function is_kvm
+{
+    [[ $(virt-what) == "kvm" ]] && return 0 || return 1
+}
+
+function startup
+{
+    is_kvm
+    if (( $? == 0 )); then
+        rlSetReason $BKRM_UNSUPPORTED "kvm is unsupported"
+        return $BKRM_UNSUPPORTED
+    fi
+
+    is_intel
+    if (( $? != 0 )); then
+        rlSetReason $BKRM_UNSUPPORTED "non-intel CPU is unsupported"
+        return $BKRM_UNSUPPORTED
+    fi
+
+    if [[ ! -d $TMPDIR ]]; then
+        rlRun "mkdir -p -m 0755 $TMPDIR" || return $BKRM_FAIL
```

Comment:
Shouln't this be abort rather then fail? Not that it would be expected to fail but it's a preparation for test, rather then part of the test itself.

Reply:
Other than this, the changes look good!

Reply:
Okay, use `return $BKRM_FATAL` instead of `return $BKRM_FAIL`. BTW, 
* Also `return $BKRM_FATAL` when executing `rlRun -l "modprobe intel-rapl"` because it is also a preparation for test;
* File `cpu/driver/runtest.sh` and `cpu/idle/runtest.sh` are updated too as they have the same code line.

Reply:
Just now I renamed `BKRM_FATAL` to be `BKRM_UNINITIATED` as word `UNINITIATED` is more precise to describe the test code that is in preparation phase before a real test is started. For details, please refer to:
* http://pastebin.test.redhat.com/782545